ethapi/api.go

```diff
@@ -694,22 +695,26 @@ func (s *PublicBlockChainAPI) GetHeaderByNumber(ctx context.Context, number rpc.
 func (s *PublicBlockChainAPI) GetHeaderByHash(ctx context.Context, hash common.Hash) map[string]interface{} {
 	header, _ := s.b.HeaderByHash(ctx, hash)
 	if header != nil {
-		return s.rpcMarshalHeader(header, s.calculateLogsBloom(ctx, rpc.BlockNumber(header.Number.Uint64())))
+		return s.rpcMarshalHeader(header, s.calculateExtBlockApi(ctx, rpc.BlockNumber(header.Number.Uint64())))
 	}
 	return nil
 }
 
-func (s *PublicBlockChainAPI) calculateLogsBloom(ctx context.Context, blkNumber rpc.BlockNumber) types.Bloom {
-	if s.b.CalcLogsBloom() && blkNumber != rpc.EarliestBlockNumber {
+func (s *PublicBlockChainAPI) calculateExtBlockApi(ctx context.Context, blkNumber rpc.BlockNumber) extBlockApi {
+	var ext extBlockApi
+	if s.b.CalcBlockExtApi() && blkNumber != rpc.EarliestBlockNumber {
 		receipts, err := s.b.GetReceiptsByNumber(ctx, blkNumber)
 		if err != nil {
-			return types.Bloom{}
+			return ext
 		}
-		if receipts != nil {
-			return types.CreateBloom(receipts)
+		if len(receipts) != 0 {
```

Comment:
nitpick: I found that `types.Receipts` already has a wrapper method to get len; so I think it'd better to use it 

```suggestion
		if receipts.Len() != 0 {
```